pygeos/test/test_constructive.py

```diff
@@ -306,3 +306,28 @@ def test_reverse_none():
 def test_reverse_invalid_type(geom):
     with pytest.raises(TypeError, match="One of the arguments is of incorrect type"):
         pygeos.reverse(geom)
+
+
+@pytest.mark.parametrize("geometry", all_types)
+def test_clip_by_rect(geometry):
+    actual = pygeos.clip_by_rect([geometry, geometry], 0.0, 0.0, 1.0, 1.0)
+    assert actual.shape == (2,)
+    assert actual[0] is None or isinstance(actual[0], Geometry)
```

Comment:
This test doesn't validate expected results from the clipping operation.  I'd recommend using specific pairs of inputs and clipped outputs.

Do watch for this case though: https://trac.osgeo.org/geos/ticket/1056
I was able to reproduce the issue on 3.8

Reply:
> This test doesn't validate expected results from the clipping operation. I'd recommend using specific pairs of inputs and clipped outputs.

I know, but that's what we do for most other functions in this test file as well (I suppose in the rationale: we test that it works, but rely for correctness on GEOS). Now, I can certainly add a test with a few cases that ensures it's doing a clipping operation and not just something random (suppose that I called the wrong GEOS function in ufuncs.c), but it's always going to be limited.

Reply:
I added some test cases from shapely!